superset-frontend/src/SqlLab/components/TableElement.tsx

```diff
@@ -92,15 +107,15 @@ const TableElement = props => {
           />
         );
       }
-      let latest = Object.entries(table.partitions?.latest || []).map(
+      const latest = Object.entries(table.partitions?.latest || []).map(
         ([key, value]) => `${key}=${value}`,
```

Comment:
is this why you made key into string | number? Because this key is a different key than the one in props. It looks like key is always a number. But also that it isn't being pushed in.